src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -573,7 +580,7 @@ function ChildReconciler(shouldClone, shouldTrackSideEffects) {
             // it from the child list so that we don't add it to the deletion
             // list.
             existingChildren.delete(
-              newFiber.key === null ? newFiber.index : newFiber.key
+              newFiber.key === null ? newIdx : newFiber.key
```

Comment:
`newFiber.index` will be `0` if this is a clone. The index we actually matched on is `oldFiber.index` which also happens to be the same as `newIdx` which is what we just looked up in the map.